index.js

```diff
@@ -373,6 +388,41 @@ function sanitizeHtml(html, options, _recursing) {
     return !options.allowedSchemes || options.allowedSchemes.indexOf(scheme) === -1;
   }
 
+  function filterCss(ast) {
+    var originalAst = ast;
+    // there should only be one element in the ast array based on how we set it up
+    ast = ast.stylesheet.rules[0];
+    // Check if selector is in our allowed styles
+    if(options.allowedStyles) {
+      if(options.allowedStyles[ast.selectors[0]] || options.allowedStyles['*']) {
+        var allowedDeclarations = [];
+        ast.declarations.forEach(function(value) {
+          // Is specific style allowed?
+          // First check if allowed in glob
+          if(options.allowedStyles['*']) {
+            if(options.allowedStyles['*'][value.property]) {
+              if(options.allowedStyles['*'][value.property] === '*' || options.allowedStyles['*'][value.property].indexOf(value.value) > -1) {
+                allowedDeclarations.push(value);
+              }
+            }
+          } else if (options.allowedStyles[ast.selectors[0]]) {
+            if(options.allowedStyles[ast.selectors[0]][value.property]) {
+              if(options.allowedStyles[ast.selectors[0]][value.property] === '*' || options.allowedStyles[ast.selectors[0]][value.property].indexOf(value.value) > -1) {
+                allowedDeclarations.push(value);
+              }
+            }
+          }
+        });
+        originalAst.stylesheet.rules[0].declarations = allowedDeclarations;
+        return originalAst;
+      } else {
+        return false;
+      }
+    } else {
+      return originalAst;
+    }
+  }
```

Comment:
This method could be broken out into several smaller functions to make it more dry. I don't mind volunteering to do it.

Reply:
Also this is where we should do validation on the values of these css properties. It's be easy to do if Regexps are part of our whitelist

Reply:
I'll refactor on postcss and leave the function as-is, and then if you wanna chop it up, go for it!

Reply:
@cwill747 sounds good!